.travis.yml

```diff
@@ -1,26 +1,92 @@
+# no compilation errors), and runs all the tests. If any of the compilation/test
+# phases fail, the build is marked as a failure.
+#
+# We handle two types of package dependencies:
+#   - packages (ros and otherwise) available through apt-get. These are installed
+#     using rosdep, based on the information in the ROS package.xml.
+#   - dependencies that must be checked out from source. These are handled by
+#     'wstool', and should be listed in a file named dependencies.rosinstall.
+#
+# There are two variables you may want to change:
+#   - ROS_DISTRO (default is indigo). Note that packages must be available for
+#     ubuntu 14.04 trusty.
+#   - ROSINSTALL_FILE (default is dependencies.rosinstall inside the repo
+#     root). This should list all necessary repositories in wstool format (see
+#     the ros wiki). If the file does not exists then nothing happens.
+#
+# See the README.md for more information.
+#
+# Author: Felix Duvallet <felixd@gmail.com>
+
+# NOTE: The build lifecycle on Travis.ci is something like this:
+#    before_install
+#    install
+#    before_script
+#    script
+#    after_success or after_failure
+#    after_script
+#    OPTIONAL before_deploy
+#    OPTIONAL deploy
+#    OPTIONAL after_deploy
 
+################################################################################
+
+# Use ubuntu trusty (14.04) with sudo privileges.
+dist: trusty
+sudo: required
 language:
-    c++
+  - python
+  - c++
+cache:
+  - apt
+
+# By default travis runs all python code in a virtualenv that does not contain
+# the packages installed with apt-get. As this is essential (pip doesn't contain
+# all the necessary ROS python packages), tell travis-ci to use the system-wide
+# python packages.
+virtualenv:
+  system_site_packages: true
+
+# Configuration variables. All variables are global now, but this can be used to
+# trigger a build matrix for different ROS distributions if desired.
+env:
+  global:
+    - ROS_DISTRO=indigo
+    - ROS_CI_DESKTOP="`lsb_release -cs`"  # e.g. [precise|trusty|...]
+    - ROS_PARALLEL_JOBS='-j8 -l6'
+
+################################################################################
 
-compiler:
-    - gcc
+# Install system dependencies, namely a very barebones ROS setup.
+before_install:
+  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+  - sudo apt-get update -qq
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin ros-$ROS_DISTRO-desktop
+  - sudo apt-get install -y ros-$ROS_DISTRO-pcl-ros ros-$ROS_DISTRO-image-transport ros-$ROS_DISTRO-image-transport-plugins ros-$ROS_DISTRO-libg2o
+    # Prepare rosdep to install dependencies.
+  - sudo rosdep init
+  - rosdep update
+  - source /opt/ros/$ROS_DISTRO/setup.bash
+  - sudo apt-get install python-rosinstall
```

Comment:
I think this is better in that it does not grab the full desktop ros, only the packages needed for mcptam (or so we say), and gets the key instead of hardcoding it in the script.  It also sets up the terminal, which is nice to have if we end up doing multiple tests going forward.  So most of these things are good, I think.